python/teams_ai/planner/planner.py

```diff
@@ -0,0 +1,39 @@
+"""
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+"""
+
+from abc import ABC, abstractclassmethod
+from typing import Any, Callable, Union
+from teams_ai.turn_state import TurnState
+
+class Planner(ABC):
+    """
+    interface implemented by all planners
+    """
+
+    @abstractclassmethod
+    async def generate_plan(self, state: TurnState, prompt_name_or_template: Union[str, PromptTemplate], *, history_options: AIHistoryOptions) -> Plan:
```

Comment:
I don't know what `turnContext` is for either. Guess it is extensible for the customized planner.

Reply:
Customized planner/moderator is one of the scenarios that may need `turnContext`.

IMO, `turnContext` is a good entry point to get teams specific context, e.g.,
- `turnContext.Activity.GetChannelData`
- `TeamsInfo.GetTeamDetailsAsync(turnContext, ...)`
- ...

It's easier to customize planner/moderator behavior per team if having `turnContext` as input.

Reply:
@swatDong that makes sense to me, I will add `turnContext` as parameter